Add dconf_gnome_disable_automount to RHEL STIG profile.

ggbecker opened the pull request:
#### Description:

- Select rule to STIG profile
- Update references section
- Add RHEL8 cce
- Enable remediation for Fedora
- Change severity to medium as defined by DISA STIG
#### Rationale:

- Rule is part of newer STIG: https://vaulted.io/library/disa-stigs-srgs/red_hat_enterprise_linux_7_security_technical_implementation_guide/V-100023?version=v2r7

Update1: Rule has been broken down into three rules, each of them checks for a specific parameter in gnome dconf configuration.

mildas commented:
**Changes identified:**
Profile stig on rhel7:
 Rule dconf_gnome_disable_automount added to stig profile.

**Recommended tests to execute:**
 build_product rhel7
 tests/test_suite.py profile --libvirt qemu:///system test-suite-vm --datastream build/ssg-rhel7-ds.xml stig

mildas commented:
**Changes identified:**
Profile stig on rhel7:
 Rule dconf_gnome_disable_automount added to stig profile.

**Recommended tests to execute:**
 build_product rhel7
 tests/test_suite.py profile --libvirt qemu:///system test-suite-vm --datastream build/ssg-rhel7-ds.xml stig

mildas commented:
**Changes identified:**
Profile stig on rhel7:
 Rule dconf_gnome_disable_automount added to stig profile.

**Recommended tests to execute:**
 build_product rhel7
 tests/test_suite.py profile --libvirt qemu:///system test-suite-vm --datastream build/ssg-rhel7-ds.xml stig

mildas commented:
**Changes identified:**
Profile stig on rhel7:
 Rule dconf_gnome_disable_automount added to stig profile.

**Recommended tests to execute:**
 build_product rhel7
 tests/test_suite.py profile --libvirt qemu:///system test-suite-vm --datastream build/ssg-rhel7-ds.xml stig

vojtapolasek commented:
I am trying to run tests for this rule with upstream test suite and VM and the rule errors out as "not applicable". I see there is
platform: machine
Is that needed? Is this the reason why the test reports "not applicable"?

ggbecker commented:
> I am trying to run tests for this rule with upstream test suite and VM and the rule errors out as "not applicable". I see there is
> platform: machine
> Is that needed? Is this the reason why the test reports "not applicable"?

You need to have `dconf` installed in your testing VM.

redhatrises commented:
This rule needs to be broken into at least 3 separate rules.

vojtapolasek commented:
> This rule needs to be broken into at least 3 separate rules.

What is your reasoning? Would it make sense to have in a profile for example only two of these three config variables configured?

redhatrises commented:
> > This rule needs to be broken into at least 3 separate rules.
> 
> What is your reasoning? Would it make sense to have in a profile for example only two of these three config variables configured?

Yes. It would. It also produces better security content.

ggbecker commented:
> > > This rule needs to be broken into at least 3 separate rules.
> > 
> > 
> > What is your reasoning? Would it make sense to have in a profile for example only two of these three config variables configured?
> 
> Yes. It would. It also produces better security content.

Ok, I'll add this to my backlog and will get back to it soon.

JAORMX commented:
/retest

mildas commented:
**Changes identified:**
Rule dconf_gnome_disable_automount:
 Template usage changed in ansible remediation.
 Attribute value changed in OVAL check.
 Ansible remediation changed.
 Node moved within OVAL check.
 Text changed in OVAL check.
 Deleted attribute from OVAL check.
 Node deleted from OVAL check.
Rule dconf_gnome_disable_automount_open:
 OVAL check is newly added.
 Ansible remediation newly added.
 Bash remediation is newly added.
Rule dconf_gnome_disable_autorun:
 OVAL check is newly added.
 Ansible remediation newly added.
 Bash remediation is newly added.
Profile ncp on rhel7:
 Rule dconf_gnome_disable_automount_open, dconf_gnome_disable_autorun added to ncp profile.
Profile stig on rhel7:
 Rule dconf_gnome_disable_automount_open, dconf_gnome_disable_automount, dconf_gnome_disable_autorun added to stig profile.

**Recommended tests to execute:**
 build_product rhel7
 tests/test_suite.py rule --libvirt qemu:///system test-suite-vm --remediate-using ansible --datastream build/ssg-rhel7-ds.xml dconf_gnome_disable_automount
 tests/test_suite.py rule --libvirt qemu:///system test-suite-vm --remediate-using bash --datastream build/ssg-rhel7-ds.xml dconf_gnome_disable_automount
 tests/test_suite.py rule --libvirt qemu:///system test-suite-vm --remediate-using ansible --datastream build/ssg-rhel7-ds.xml dconf_gnome_disable_automount_open
 tests/test_suite.py rule --libvirt qemu:///system test-suite-vm --remediate-using bash --datastream build/ssg-rhel7-ds.xml dconf_gnome_disable_automount_open
 tests/test_suite.py rule --libvirt qemu:///system test-suite-vm --remediate-using ansible --datastream build/ssg-rhel7-ds.xml dconf_gnome_disable_autorun
 tests/test_suite.py rule --libvirt qemu:///system test-suite-vm --remediate-using bash --datastream build/ssg-rhel7-ds.xml dconf_gnome_disable_autorun
 tests/test_suite.py profile --libvirt qemu:///system test-suite-vm --datastream build/ssg-rhel7-ds.xml ncp
 tests/test_suite.py profile --libvirt qemu:///system test-suite-vm --datastream build/ssg-rhel7-ds.xml stig

ggbecker commented:
> > > > This rule needs to be broken into at least 3 separate rules.
> > > 
> > > 
> > > What is your reasoning? Would it make sense to have in a profile for example only two of these three config variables configured?
> > 
> > 
> > Yes. It would. It also produces better security content.
> 
> Ok, I'll add this to my backlog and will get back to it soon.

@redhatrises rule has been broken down into three rules. Can you verify them?

mildas commented:
**Changes identified:**
Rule dconf_gnome_disable_automount:
 Deleted attribute from OVAL check.
 Node moved within OVAL check.
 Attribute value changed in OVAL check.
 Ansible remediation changed.
 Text changed in OVAL check.
 Node deleted from OVAL check.
 Template usage changed in ansible remediation.
Rule dconf_gnome_disable_automount_open:
 Bash remediation is newly added.
 Ansible remediation newly added.
 OVAL check is newly added.
Rule dconf_gnome_disable_autorun:
 Bash remediation is newly added.
 Ansible remediation newly added.
 OVAL check is newly added.
Profile ncp on rhel7:
 Rule dconf_gnome_disable_automount_open, dconf_gnome_disable_autorun added to ncp profile.
Profile stig on rhel7:
 Rule dconf_gnome_disable_automount, dconf_gnome_disable_automount_open, dconf_gnome_disable_autorun added to stig profile.

**Recommended tests to execute:**
 build_product rhel7
 tests/test_suite.py rule --libvirt qemu:///system test-suite-vm --remediate-using ansible --datastream build/ssg-rhel7-ds.xml dconf_gnome_disable_automount
 tests/test_suite.py rule --libvirt qemu:///system test-suite-vm --remediate-using bash --datastream build/ssg-rhel7-ds.xml dconf_gnome_disable_automount
 tests/test_suite.py rule --libvirt qemu:///system test-suite-vm --remediate-using ansible --datastream build/ssg-rhel7-ds.xml dconf_gnome_disable_automount_open
 tests/test_suite.py rule --libvirt qemu:///system test-suite-vm --remediate-using bash --datastream build/ssg-rhel7-ds.xml dconf_gnome_disable_automount_open
 tests/test_suite.py rule --libvirt qemu:///system test-suite-vm --remediate-using ansible --datastream build/ssg-rhel7-ds.xml dconf_gnome_disable_autorun
 tests/test_suite.py rule --libvirt qemu:///system test-suite-vm --remediate-using bash --datastream build/ssg-rhel7-ds.xml dconf_gnome_disable_autorun
 tests/test_suite.py profile --libvirt qemu:///system test-suite-vm --datastream build/ssg-rhel7-ds.xml ncp
 tests/test_suite.py profile --libvirt qemu:///system test-suite-vm --datastream build/ssg-rhel7-ds.xml stig

mildas commented:
**Changes identified:**
Rule dconf_gnome_disable_automount:
 Ansible remediation changed.
 Deleted attribute from OVAL check.
 Attribute value changed in OVAL check.
 Text changed in OVAL check.
 Node moved within OVAL check.
 Node deleted from OVAL check.
 Template usage changed in ansible remediation.
Rule dconf_gnome_disable_automount_open:
 Ansible remediation newly added.
 Bash remediation is newly added.
 OVAL check is newly added.
Rule dconf_gnome_disable_autorun:
 Ansible remediation newly added.
 Bash remediation is newly added.
 OVAL check is newly added.
Profile ncp on rhel7:
 Rule dconf_gnome_disable_automount_open, dconf_gnome_disable_autorun added to ncp profile.
Profile stig on rhel7:
 Rule dconf_gnome_disable_automount, dconf_gnome_disable_automount_open, dconf_gnome_disable_autorun added to stig profile.

**Recommended tests to execute:**
 build_product rhel7
 tests/test_suite.py rule --libvirt qemu:///system test-suite-vm --remediate-using ansible --datastream build/ssg-rhel7-ds.xml dconf_gnome_disable_automount
 tests/test_suite.py rule --libvirt qemu:///system test-suite-vm --remediate-using bash --datastream build/ssg-rhel7-ds.xml dconf_gnome_disable_automount
 tests/test_suite.py rule --libvirt qemu:///system test-suite-vm --remediate-using ansible --datastream build/ssg-rhel7-ds.xml dconf_gnome_disable_automount_open
 tests/test_suite.py rule --libvirt qemu:///system test-suite-vm --remediate-using bash --datastream build/ssg-rhel7-ds.xml dconf_gnome_disable_automount_open
 tests/test_suite.py rule --libvirt qemu:///system test-suite-vm --remediate-using ansible --datastream build/ssg-rhel7-ds.xml dconf_gnome_disable_autorun
 tests/test_suite.py rule --libvirt qemu:///system test-suite-vm --remediate-using bash --datastream build/ssg-rhel7-ds.xml dconf_gnome_disable_autorun
 tests/test_suite.py profile --libvirt qemu:///system test-suite-vm --datastream build/ssg-rhel7-ds.xml ncp
 tests/test_suite.py profile --libvirt qemu:///system test-suite-vm --datastream build/ssg-rhel7-ds.xml stig

ggbecker commented:
@redhatrises can you check this one?

mildas commented:
**Changes identified:**
Rule dconf_gnome_disable_automount:
 Attribute value changed in OVAL check.
 Ansible remediation changed.
 Node moved within OVAL check.
 Deleted attribute from OVAL check.
 Node deleted from OVAL check.
 Template usage changed in ansible remediation.
 Text changed in OVAL check.
Rule dconf_gnome_disable_automount_open:
 OVAL check is newly added.
 Bash remediation is newly added.
 Ansible remediation newly added.
Rule dconf_gnome_disable_autorun:
 OVAL check is newly added.
 Bash remediation is newly added.
 Ansible remediation newly added.
Profile ncp on rhel7:
 Rule dconf_gnome_disable_automount_open, dconf_gnome_disable_autorun added to ncp profile.
Profile stig on rhel7:
 Rule dconf_gnome_disable_automount, dconf_gnome_disable_automount_open, dconf_gnome_disable_autorun added to stig profile.

**Recommended tests to execute:**
 build_product rhel7
 tests/test_suite.py rule --libvirt qemu:///system test-suite-vm --remediate-using ansible --datastream build/ssg-rhel7-ds.xml dconf_gnome_disable_automount
 tests/test_suite.py rule --libvirt qemu:///system test-suite-vm --remediate-using bash --datastream build/ssg-rhel7-ds.xml dconf_gnome_disable_automount
 tests/test_suite.py rule --libvirt qemu:///system test-suite-vm --remediate-using ansible --datastream build/ssg-rhel7-ds.xml dconf_gnome_disable_automount_open
 tests/test_suite.py rule --libvirt qemu:///system test-suite-vm --remediate-using bash --datastream build/ssg-rhel7-ds.xml dconf_gnome_disable_automount_open
 tests/test_suite.py rule --libvirt qemu:///system test-suite-vm --remediate-using ansible --datastream build/ssg-rhel7-ds.xml dconf_gnome_disable_autorun
 tests/test_suite.py rule --libvirt qemu:///system test-suite-vm --remediate-using bash --datastream build/ssg-rhel7-ds.xml dconf_gnome_disable_autorun
 tests/test_suite.py profile --libvirt qemu:///system test-suite-vm --datastream build/ssg-rhel7-ds.xml ncp
 tests/test_suite.py profile --libvirt qemu:///system test-suite-vm --datastream build/ssg-rhel7-ds.xml stig

redhatrises commented:
Looks good except for the duplicate STIGIDs. We should remove them, and ask for new ones.

ggbecker commented:
> Looks good except for the duplicate STIGIDs. We should remove them, and ask for new ones.

I assume that it can take a while until they (DISA) split the rules and assign new STIG ids to them. So, do we want to wait for that? And how do we initiate this request to DISA?

vojtapolasek commented on linux_os/guide/system/software/gnome/gnome_media_settings/dconf_gnome_disable_automount_open/oval/shared.xml:
```diff
@@ -0,0 +1,50 @@
+    <metadata>
+      <title>Disable GNOME3 automount-open</title>
+      <affected family="unix">
+        <platform>Red Hat Enterprise Linux 7</platform>
```
I would use the Jinja macro here.

ggbecker replied:
Used `oval_metadata` macro.

vojtapolasek commented on linux_os/guide/system/software/gnome/gnome_media_settings/dconf_gnome_disable_automount/oval/shared.xml:
```diff
@@ -1,27 +1,22 @@
     <metadata>
-      <title>Disable GNOME3 Automounting</title>
+      <title>Disable GNOME3 automount</title>
       <affected family="unix">
```
I would use Jinja macro here.

ggbecker replied:
Used `oval_metadata` macro.

vojtapolasek commented on linux_os/guide/system/software/gnome/gnome_media_settings/dconf_gnome_disable_autorun/oval/shared.xml:
```diff
@@ -0,0 +1,50 @@
+  <definition class="compliance" id="dconf_gnome_disable_autorun" version="1">
+    <metadata>
+      <title>Disable GNOME3 Automounting</title>
+      <affected family="unix">
```
I would use Jinja macro here.

ggbecker replied:
Used `oval_metadata` macro.

vojtapolasek commented on linux_os/guide/system/software/gnome/gnome_media_settings/dconf_gnome_disable_automount_open/oval/shared.xml:
```diff
@@ -0,0 +1,50 @@
+  version="1">
+    <ind:path>/etc/dconf/db/local.d/</ind:path>
+    <ind:filename operation="pattern match">^.*$</ind:filename>
+    <ind:pattern operation="pattern match">^\[org/gnome/desktop/media-handling]([^\n]*\n+)+?automount-open=false$</ind:pattern>
```
I see that the  closing bracket is not escaped... it is not causing any problems now but I would do that. Please see also other ovals.
```suggestion
    <ind:pattern operation="pattern match">^\[org/gnome/desktop/media-handling\]([^\n]*\n+)+?automount-open=false$</ind:pattern>
```

ggbecker replied:
Done

vojtapolasek commented on linux_os/guide/system/software/gnome/gnome_media_settings/dconf_gnome_disable_autorun/oval/shared.xml:
```diff
@@ -0,0 +1,50 @@
+  version="1">
+    <ind:path>/etc/dconf/db/local.d/</ind:path>
+    <ind:filename operation="pattern match">^.*$</ind:filename>
+    <ind:pattern operation="pattern match">^\[org/gnome/desktop/media-handling]([^\n]*\n+)+?autorun-never=true$</ind:pattern>
```
```suggestion
    <ind:pattern operation="pattern match">^\[org/gnome/desktop/media-handling\]([^\n]*\n+)+?autorun-never=true$</ind:pattern>
```

ggbecker replied:
Done

vojtapolasek reviewed (changes requested):
Looks very good, just small fixes needed, please see comments.

vojtapolasek commented:
Oh, while running SST test suite, unfortunately none of ansible remediations seem to pass the test. Bash is ok.

ggbecker commented:
> Oh, while running SST test suite, unfortunately none of ansible remediations seem to pass the test. Bash is ok.

I'll eventually get back to this pull request when I have time.

mildas commented:
**Changes identified:**
Rule dconf_gnome_disable_automount:
 Node moved within OVAL check.
 Ansible remediation changed.
 Node deleted from OVAL check.
 Attribute value changed in OVAL check.
 Text changed in OVAL check.
 Template usage changed in ansible remediation.
 Deleted attribute from OVAL check.
Rule dconf_gnome_disable_automount_open:
 OVAL check is newly added.
 Ansible remediation newly added.
 Bash remediation is newly added.
Rule dconf_gnome_disable_autorun:
 OVAL check is newly added.
 Ansible remediation newly added.
 Bash remediation is newly added.
Profile ncp on rhel7:
 Rule dconf_gnome_disable_automount_open, dconf_gnome_disable_autorun added to ncp profile.
Profile stig on rhel7:
 Rule dconf_gnome_disable_automount_open, dconf_gnome_disable_autorun, dconf_gnome_disable_automount added to stig profile.

**Recommended tests to execute:**
 build_product rhel7
 tests/test_suite.py rule --libvirt qemu:///system test-suite-vm --remediate-using ansible --datastream build/ssg-rhel7-ds.xml dconf_gnome_disable_automount
 tests/test_suite.py rule --libvirt qemu:///system test-suite-vm --remediate-using bash --datastream build/ssg-rhel7-ds.xml dconf_gnome_disable_automount
 tests/test_suite.py rule --libvirt qemu:///system test-suite-vm --remediate-using ansible --datastream build/ssg-rhel7-ds.xml dconf_gnome_disable_automount_open
 tests/test_suite.py rule --libvirt qemu:///system test-suite-vm --remediate-using bash --datastream build/ssg-rhel7-ds.xml dconf_gnome_disable_automount_open
 tests/test_suite.py rule --libvirt qemu:///system test-suite-vm --remediate-using ansible --datastream build/ssg-rhel7-ds.xml dconf_gnome_disable_autorun
 tests/test_suite.py rule --libvirt qemu:///system test-suite-vm --remediate-using bash --datastream build/ssg-rhel7-ds.xml dconf_gnome_disable_autorun
 tests/test_suite.py profile --libvirt qemu:///system test-suite-vm --datastream build/ssg-rhel7-ds.xml ncp
 tests/test_suite.py profile --libvirt qemu:///system test-suite-vm --datastream build/ssg-rhel7-ds.xml stig

ggbecker commented:
> Oh, while running SST test suite, unfortunately none of ansible remediations seem to pass the test. Bash is ok.

I have addressed all the requests and fixed the ansible remediation (it was missing `no_extra_spaces: yes`). The OVAL regex doesn't expect any space between the `=` separator.

redhatrises commented:
> > Looks good except for the duplicate STIGIDs. We should remove them, and ask for new ones.
> 
> I assume that it can take a while until they (DISA) split the rules and assign new STIG ids to them. So, do we want to wait for that? And how do we initiate this request to DISA?

Yes, we do want to wait for that. There should be no duplicate STIG IDs.

ggbecker commented:
> > > Looks good except for the duplicate STIGIDs. We should remove them, and ask for new ones.
> > 
> > 
> > I assume that it can take a while until they (DISA) split the rules and assign new STIG ids to them. So, do we want to wait for that? And how do we initiate this request to DISA?
> 
> Yes, we do want to wait for that. There should be no duplicate STIG IDs.

Ok, also I remember seeing duplicated STIG ids in the project. I will try to find all of them and publish in a new issue.

redhatrises commented:
/retest

redhatrises commented on linux_os/guide/system/software/gnome/gnome_media_settings/dconf_gnome_disable_automount_open/rule.yml:
```diff
@@ -0,0 +1,57 @@
+    cobit5: APO13.01,DSS01.04,DSS05.03,DSS05.04,DSS05.05,DSS05.07,DSS06.03
+    iso27001-2013: A.11.2.6,A.13.1.1,A.13.2.1,A.6.2.1,A.6.2.2,A.7.1.1,A.9.2.1
+    cis-csc: 12,16
+    stig@rhel7: RHEL-07-020111
```
Remove this duplicate ID.

ggbecker replied:
Done

redhatrises commented on linux_os/guide/system/software/gnome/gnome_media_settings/dconf_gnome_disable_autorun/rule.yml:
```diff
@@ -0,0 +1,57 @@
+    cobit5: APO13.01,DSS01.04,DSS05.03,DSS05.04,DSS05.05,DSS05.07,DSS06.03
+    iso27001-2013: A.11.2.6,A.13.1.1,A.13.2.1,A.6.2.1,A.6.2.2,A.7.1.1,A.9.2.1
+    cis-csc: 12,16
+    stig@rhel7: RHEL-07-020111
```
Remove this duplicate ID.

ggbecker replied:
Done
